src/fast_scanning.jl

```diff
@@ -0,0 +1,136 @@
+sharpness{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}) = var(imfilter(img, Kernel.Laplacian(ntuple(i->true, Val{N}))))
+
+function adaptive_thres{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}, block::NTuple{N,Int})
+    threshold = zeros(Float64, block)
+    block_length = CartesianIndex(ntuple(i->ceil(Int,length(indices(img,i))/block[i]),Val{N}))
+    net_s = sharpness(img)
+    net_var = var(img)
+    net_end = last(CartesianRange(indices(img)))
+    for i in CartesianRange(block)
+        si = CartesianIndex(ntuple(j->(i[j]-1)*block_length[j]+1,Val{N}))
+        ei = min(si + block_length - 1, net_end)
+        wi = view(img, map((i,j)->i:j, si.I, ei.I)...)
+        threshold[i] = 0.02 + min(sharpness(wi)/net_s*0.04,0.1) + min(var(wi)/net_var*0.04,0.1)
+    end
+    threshold
+end
+
+getscalar{T<:Real,N}(A::AbstractArray{T,N}, i::CartesianIndex{N}, block_length::CartesianIndex{N}) =
+    A[CartesianIndex(ntuple(j->(i[j]-1)÷block_length[j]+1, Val{N}))]
+
+getscalar(a::Real, i...) = a
+
+"""
+    seg_img = fast_scanning(img, threshold, [diff_fn])
+
+Segments the N-D image using a fast scanning algorithm and returns a
+[`SegmentedImage`](@ref) containing information about the segments.
+
+# Arguments:
+* `img`         : N-D image to be segmented (arbitrary indices are allowed)
+* `threshold`   : Upper bound of the difference measure (δ) for considering
+                  pixel into same segment; an `AbstractArray` can be passed
+                  having same number of dimensions as that of `img` for adaptive
+                  thresholding
+* `diff_fn`     : (Optional) Function that returns a difference measure (δ)
+                  between the mean color of a region and color of a point
+
+# Examples:
+
+```jldoctest
+julia> img = zeros(Float64, (3,3));
+julia> img[2,:] = 0.5;
+julia> img[:,2] = 0.6;
+julia> seg = fast_scanning(img, 0.2);
+julia> seg.image_indexmap
+3×3 Array{Int64,2}:
+ 1  4  5
+ 4  4  4
+ 3  4  6
+
+```
+"""
+fast_scanning{CT<:Images.NumberLike,N}(img::AbstractArray{CT,N}, block::NTuple{N,Int} = ntuple(i->4,Val{N})) = fast_scanning(img, adaptive_thres(img, block))
+
+function fast_scanning{CT<:Union{Colorant,Real},N}(img::AbstractArray{CT,N}, threshold::Union{AbstractArray,Real}, diff_fn::Function = default_diff_fn)
+
+    if typeof(threshold) <: AbstractArray
```

Comment:
Either is fine by me. If you want to drop 0.5 just make sure you change the REQUIRE and CI scripts appropriately.